src/aggregation_merkle_sum_tree.rs

```diff
@@ -1,45 +1,57 @@
 use halo2_proofs::halo2curves::bn256::Fr as Fp;
 use num_bigint::BigUint;
+use std::error::Error;
 use summa_backend::merkle_sum_tree::utils::{build_merkle_tree_from_leaves, fp_to_big_uint};
-use summa_backend::merkle_sum_tree::{Entry, MerkleProof, MerkleSumTree, Node, Tree};
+use summa_backend::merkle_sum_tree::{
+    Cryptocurrency, Entry, MerkleProof, MerkleSumTree, Node, Tree,
+};
 
 /// Aggregation Merkle Sum Tree Data Structure.
 ///
-/// Starting from a set of "mini" Merkle Sum Trees of equal depth, N_ASSETS and N_BYTES, the Aggregation Merkle Sum Tree inherits the properties of a Merkle Sum Tree and adds the following:
+/// Starting from a set of "mini" Merkle Sum Tree of equal depth, N_CURRENCIES and N_BYTES, the Aggregation Merkle Sum Tree inherits the properties of a Merkle Sum Tree and adds the following:
 /// * Each Leaf of the Aggregation Merkle Sum Tree is the root of a "mini" Merkle Sum Tree made of `hash` and `balances`
 ///
 /// # Type Parameters
 ///
-/// * `N_ASSETS`: The number of assets for each user account
+/// * `N_CURRENCIES`: The number of assets for each user account
 /// * `N_BYTES`: Range in which each node balance should lie
 #[derive(Debug, Clone)]
-pub struct AggregationMerkleSumTree<const N_ASSETS: usize, const N_BYTES: usize> {
-    root: Node<N_ASSETS>,
-    nodes: Vec<Vec<Node<N_ASSETS>>>,
+pub struct AggregationMerkleSumTree<const N_CURRENCIES: usize, const N_BYTES: usize> {
+    root: Node<N_CURRENCIES>,
+    nodes: Vec<Vec<Node<N_CURRENCIES>>>,
     depth: usize,
-    mini_trees: Vec<MerkleSumTree<N_ASSETS, N_BYTES>>,
+    cryptocurrencies: Vec<Cryptocurrency>,
+    mini_trees: Vec<MerkleSumTree<N_CURRENCIES, N_BYTES>>,
 }
 
-impl<const N_ASSETS: usize, const N_BYTES: usize> Tree<N_ASSETS, N_BYTES>
-    for AggregationMerkleSumTree<N_ASSETS, N_BYTES>
+impl<const N_CURRENCIES: usize, const N_BYTES: usize> Tree<N_CURRENCIES, N_BYTES>
+    for AggregationMerkleSumTree<N_CURRENCIES, N_BYTES>
 {
-    fn root(&self) -> &Node<N_ASSETS> {
+    fn root(&self) -> &Node<N_CURRENCIES> {
         &self.root
     }
 
     fn depth(&self) -> &usize {
         &self.depth
     }
 
-    fn leaves(&self) -> &[Node<N_ASSETS>] {
+    fn leaves(&self) -> &[Node<N_CURRENCIES>] {
         &self.nodes[0]
     }
 
-    fn nodes(&self) -> &[Vec<Node<N_ASSETS>>] {
+    fn nodes(&self) -> &[Vec<Node<N_CURRENCIES>>] {
         &self.nodes
     }
 
-    fn get_entry(&self, user_index: usize) -> &Entry<N_ASSETS> {
+    fn cryptocurrencies(&self) -> &[Cryptocurrency] {
+        &self.cryptocurrencies
+    }
+
+    fn entries(&self) -> &[Entry<N_CURRENCIES>] {
+        self.mini_trees[0].entries()
```

Comment:
Does this mean that you are extracting the entries only from the first `mini_tree` ?

Reply:
Yes, that's correct. extract entries only from the first mini_tree to comply with the `Tree` trait requirements.
I don't think it's good idea for extracting all entries from each `mini trees` literally, because summa-aggregation will handle large size of tree.

Reply:
Ok, but this is not the behavior you would expect from the `entries()` function.  If we don't need `entries()` on the Aggregation Tree, is there a way to leave the method empty?

Reply:
> Ok, but this is not the behavior you would expect from the `entries()` function. If we don't need `entries()` on the Aggregation Tree, is there a way to leave the method empty?

Yes, one option is to remove the uncommon functions used in both `MerkleSumTree` and `AggregationMerkleSumTree` from the Tree trait.

Reply:
I propose just to return an empty array `&[]` and specify the reason of that in the comments.